pallets/public-credentials/src/lib.rs

```diff
@@ -0,0 +1,438 @@
+			);
+
+			// Check that enough funds can be reserved to pay for both attestation and
+			// public info deposits.
+			// It is harder to use two potentially different currencies while making sure
+			// that, if the same, the sum can be reserved, but if they are not, then each
+			// deposit could be reserved separately. We could switch to using
+			// `NamedReservableCurrency` to do that and check whether the name of the
+			// attestation currency matches the name of this pallet currency.
+			ensure!(
+				<CurrencyOf<T> as ReservableCurrency<T::AccountId>>::can_reserve(
+					&payer,
+					deposit_amount.saturating_add(attestation_deposit_amount)
+				),
+				Error::<T>::UnableToPayFees
+			);
+
+			// Check the validity of the claimer's signature, if present, over the
+			// credential root hash.
+			if let Some(ClaimerSignatureInfo {
+				claimer_id,
+				signature_payload,
+			}) = claimer_signature
+			{
+				T::ClaimerSignatureVerification::verify(&claimer_id, &claim_hash.encode(), &signature_payload)
+					.map_err(|err| match err {
+						SignatureVerificationError::SignerInformationNotPresent => Error::<T>::ClaimerInfoNotFound,
+						SignatureVerificationError::SignatureInvalid => Error::<T>::InvalidClaimerSignature,
+					})?;
+			}
+
+			// Delegate to the attestation pallet writing the attestation information and
+			// reserve its part of the deposit
+			attestation::Pallet::<T>::write_attestation(
+				ctype_hash,
+				claim_hash,
+				attester,
+				payer.clone(),
+				authorization_info,
+			)?;
+
+			// *** No Fail beyond this point ***
+
+			// Take the rest of the deposit. Should never fail since we made sure above that
+			// enough funds can be reserved.
+			let deposit = kilt_support::reserve_deposit::<T::AccountId, CurrencyOf<T>>(payer, deposit_amount)
+				.map_err(|_| Error::<T>::InternalError)?;
+
+			let block_number = frame_system::Pallet::<T>::block_number();
+
+			Credentials::<T>::insert(&subject, &claim_hash, CredentialEntryOf { deposit, block_number });
+			CredentialsUnicityIndex::<T>::insert(&claim_hash, subject.clone());
+
+			Self::deposit_event(Event::CredentialStored {
+				subject_id: subject,
+				claim_hash,
+			});
+
+			Ok(())
+		}
+
+		/// Removes the information pertaining a public credential from the
+		/// chain.
+		///
+		/// The preconditions of the `attestation::remove()` function must all
+		/// be fulfilled, as this function calls `attestation::remove()`
+		/// internally. Nevertheless, the opposite is not true.
+		/// Removing an attestation from the attestation pallet still requires
+		/// the attester to also remove any traces of the corresponding public
+		/// credential from this pallet.
+		///
+		/// The removal of the credential does not delete it entirely from the
+		/// blockchain history, but only its link *from* the blockchain state
+		/// *to* the blockchain history is removed.
+		///
+		/// Clients parsing public credentials should consider should interpret
```

Comment:
Probably typo?

```suggestion
		/// Clients parsing public credentials should consider to interpret
```

Reply:
Yeah seems long URLs are breaking the formatter -> https://github.com/KILTprotocol/kilt-node/pull/378/commits/3413d2bc1df24ef93cb8cc194e24a6790a19715a

Reply:
😇 https://github.com/KILTprotocol/kilt-node/pull/378/commits/90597335d7a13f27b4d6be3907c8d2a51b93840d